lib/main.js

```diff
@@ -14,16 +14,21 @@ class TypeScriptLanguageClient extends AutoLanguageClient {
     return atom.config.get('ide-typescript.javascriptSupport') ? allScopes : tsScopes
   }
   getLanguageName () { return 'TypeScript' }
-  getServerName () { return 'SourceGraph' }
+  getServerName () { return 'Theia' }
 
   startServerProcess () {
     this.supportedExtensions = atom.config.get('ide-typescript.javascriptSupport') ? allExtensions : tsExtensions
-    const args = [ 'node_modules/javascript-typescript-langserver/lib/language-server-stdio' ]
-    return super.spawnChildNode(args, { cwd: path.join(__dirname, '..') })
+    return super.spawnChildNode([
+      'node_modules/typescript-language-server/lib/cli',
+      '--stdio',
+      '--tsserver-path', this.getTypescriptServerPath()
+    ] , { cwd: path.join(__dirname, '..') })
   }
 
-  preInitialization (connection) {
-    connection.onCustom('$/partialResult', () => {}) // Suppress partialResult until the language server honors 'streaming' detection
+  getTypescriptServerPath() {
+    const tsServerPath = atom.config.get('ide-typescript.typescriptServerPath')
+    if (tsServerPath) return tsServerPath
+    return process.platform === 'win32' ? 'tsserver.cmd' : 'tsserver'
```

Comment:
Since `tsserver` has a shebang at the top the electron built in version of node is not used with `tsserver`. None of this is merged in yet but a interesting detail of the PR most likely to be merged: https://github.com/theia-ide/typescript-language-server/pull/88 is that it will only fork on files ending with `.js`, otherwise it will use `child_process.exec` on them (which will use the globally installed node version if it can correctly find it). Implementation wise I think this would mean to just use the `lib/tsserver.js` file that `tsserver` links to but for user configuration it might get interesting...

Reply:
See: https://github.com/theia-ide/typescript-language-server/pull/88#issuecomment-436850110
Since this PR got merged in the config option for `typescriptServerPath` can be removed and it can always be `node_modules/typescript/lib/tsserver.js as long as the [typescript](https://www.npmjs.com/package/typescript) package is a dependency. 

Since a `.js` file is being invoked theia will start `tsserver` by forking the current node process (and will use the node version built into electron). Additionally I don't think there needs to be anything special done on windows anymore since hashbang isn't being relied on anymore (but haven't tested this).

Reply:
By keeping this there though they can optionally use a different version of typescript from the one the Theia language server ships with?  Is that even supported that end tho?

Reply:
Yes it is, but it will only be opened "properly" (with forking logic) if the file ends with .js

What's interesting is since the bundled version of `typescript` is in the top level of the repository and they actually publish the `/server` folder, the published version of `typescript-language-server` on npm does not include the `typescript` package so it would need to be included by `ide-typescript`. See https://github.com/theia-ide/typescript-language-server/blob/master/server/package.json

Reply:
Oh, I see, they don't include typescript with their language server so we have to include it and specify path to it.  We could still let them override the path via a config setting but would need to change the default path to point to the one we bundle.

Reply:
Exactly